src/MEDS_transforms/extract/split_and_shard_patients.py

```diff
@@ -76,6 +76,16 @@ def shard_patients[
         Traceback (most recent call last):
             ...
         ValueError: Unable to adjust splits to ensure all splits have at least 1 patient.
+        >>> external_splits = {
+        ...     'train': np.array([1, 2, 3, 4, 5, 6], dtype=int),
+        ...     'test': np.array([7, 8, 9, 10], dtype=int),
+        ... }
+        >>> shard_patients(patients, 6, external_splits)
+        {'train/0': [1, 2, 3, 4, 5, 6], 'test/0': [7, 8, 9, 10]}
+        >>> shard_patients(patients, 3, external_splits)
+        Traceback (most recent call last):
+            ...
+        ValueError: External splits must have fewer patients than n_patients_per_shard (3): len(train)=6, ...
```

Comment:
You don't want it to just re-shard the external splits? Why can't it re-shard the external splits?

Reply:
You're right. It should be able to reshard them in this case. So far, it's been breaking on this, I will fix that.